File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    count = 0
    
    while len(scoville) > 1 and scoville[0] < K:
        # 2개 이상이고 K 보다 작은 것들이 있으면 섞기
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        mixed = first + (second * 2)
        heapq.heappush(scoville, mixed)
        count += 1
    
    if scoville[0] < K:
        return -1

    return count

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int s : scoville){
            pq.add(s);
        }
        
        int answer = 0;
        while(pq.size() > 1 && pq.peek() < K){
            int first = pq.poll();
            int second = pq.poll();
            int next = first + (second * 2);
            pq.add(next);
            answer++;
        }
        
        if(pq.peek() < K) return -1;
        
        return answer;
    }
}